### ISSUE TITLE
GCC 32Bit Support

### ISSUE THREAD

**Issue by AlbertShown** [civetweb/civetweb#1311]
It seems Civetweb does not compile using a 32Bit GCC compiler on Windows. Why?

_Original issue: https://github.com/webui-dev/webui/issues/532_

```
C:\Users\uia77586\Desktop\WebUi\webui>mingw32-make

C:/Users/uia77586/Desktop/WebUi/webui//src/civetweb/civetweb.c:711:8: error: redefinition of 'struct timespec'
 struct timespec {
        ^~~~~~~~
In file included from C:/Users/uia77586/Desktop/WebUi/webui//src/civetweb/civetweb.c:534:0:
c:\mingw\include\time.h:115:8: note: originally defined here
 struct timespec
        ^~~~~~~~
C:/Users/uia77586/Desktop/WebUi/webui//src/civetweb/civetweb.c:987:1: error: static declaration of 'timegm' follows non-static declaration
 timegm(struct tm *tm)
 ^~~~~~
C:/Users/uia77586/Desktop/WebUi/webui//src/civetweb/civetweb.c:671:8: note: previous declaration of 'timegm' was here
 time_t timegm(struct tm *tm);
        ^~~~~~
C:/Users/uia77586/Desktop/WebUi/webui//src/civetweb/civetweb.c: In function 'mg_read_inner':
C:/Users/uia77586/Desktop/WebUi/webui//src/civetweb/civetweb.c:6635:23: error: 'INT_MAX' undeclared (first use in this function)
      (int64_t)((len > INT_MAX) ? INT_MAX : len); /* since the return value is
                       ^~~~~~~
C:/Users/uia77586/Desktop/WebUi/webui//src/civetweb/civetweb.c:6635:23: note: each undeclared identifier is reported only once for each function it appears in
C:/Users/uia77586/Desktop/WebUi/webui//src/civetweb/civetweb.c: In function 'mg_read':
C:/Users/uia77586/Desktop/WebUi/webui//src/civetweb/civetweb.c:6753:12: error: 'INT_MAX' undeclared (first use in this function)
  if (len > INT_MAX) {
            ^~~~~~~
C:/Users/uia77586/Desktop/WebUi/webui//src/civetweb/civetweb.c: In function 'mg_write':
C:/Users/uia77586/Desktop/WebUi/webui//src/civetweb/civetweb.c:6924:12: error: 'INT_MAX' undeclared (first use in this function)
  if (len > INT_MAX) {
            ^~~~~~~
C:/Users/uia77586/Desktop/WebUi/webui//src/civetweb/civetweb.c: In function 'set_ports_option':
C:/Users/uia77586/Desktop/WebUi/webui//src/civetweb/civetweb.c:16244:29: error: 'INT_MAX' undeclared (first use in this function)
   if ((opt_listen_backlog > INT_MAX) || (opt_listen_backlog < 1)) {
                             ^~~~~~~
```

**Comment by bel2125** [comment]
It seems my last answer got lost ???

On Windows, all tests are done with Microsoft compilers or compatible ones (Intel compiler). MinGW sometimes does not follow the "Microsoft standard", but uses POSIX instead. Microsoft compilers respect POSIX only to some degree but never tried to be fully compatible. 

CivetWeb may have a few checks "#ifdef _WIN32" assuming there is a Microsoft compiler in use.
Therefore some symbols might be duplicate of missing if you use MinGW.

It should not be hard to fix, but MinGW 32-bit is not in  a regression test for Windoes.
